### DISCUSSION TITLE
Itemlevel of Drops = Monsterlevel on normal difficulty --> Intended?

### DISCUSSION THREAD

**Discussion by toastmartin** [diasurgical/DevilutionX#6892] [Ideas]
Hello, i am still enjoying the game, but wondering if the fact that monsters drop items with an item level equal to their monsterlevel on **normal** difficulty ist a bug. Probably this was unimportant to the developers, because NG+ was unimportant.
Im am playing the game more like modern Diablo games (collecting, optimizing gear). And a am always very sad to step into the next difficulty level just to get rewarded with a "Sword of Magic" (itemlevel 1) for killing a level 32 (!) Ghoul (in Hell Mode).
Do you consider the use of the unmodified Monster levels for drop generation a bug?
Would it be technically complicated to fix this?

**Comment by AJenbo** [comment]
It's a know issue and one that we are looking into solving

**Comment by DakkJaniels** [comment]
It's not a bug. The game isn't really set up well to have adjusted monster level used for affix generation.

**Reply by toastmartin** [reply]
What do you mean with "set up well"? What would the downsides of the change be? The average quality of drops would increase for nightmare and hell, especially the earlier levels.
Also, the change would cause the monsters to drop Items with suffix/prefix qlevels above 30, witch would make holy/godly, mammothe/whale, mercyless, .. actually attainable without running to Wirt. Imagine: The best drops would come from monsters, not from Merchants!

**Reply by StephenCWills** [reply]
> What would the downsides of the change be?

A much smaller pool of affixes to choose from for high-level items. Take a look at the distribution.

![image-1](https://github.com/diasurgical/devilutionX/assets/9203145/e292c904-1836-41f2-904c-626beb748b2a)

Now imagine killing Lazarus on Hell difficulty and being rewarded with a non-magical staff because the pool of suffixes dropped to zero. Does that make it sound like the quality of your drops has increased?

**Reply by DakkJaniels** [reply]
> The average quality of drops would increase for nightmare and hell, especially the earlier levels.

From a reward for effort perspective, that also seems kind of bad. Killing trash mobs in hell church would get you some of the best jewelry in the game (dragon or obsidian/zodiac). With the hp bug fix for nightmare and hell, maybe it's not as bad as it was, but regardless those enemies are not difficult in hell.

My personal opinion is that fixing this veers off into mod territory, as it's not a straightforward fix and likely will depend more upon the opinion of what someone thinks the game should be like than something that was clearly incorrect in the code.  Also, to make it work well, there should be more affixes between 35 and 60 - adding affixes would definitely be mod territory.

I like how it was done in Middle Earth mod. The affixes are still based on the unmodified monster level, but they added base items that could only drop in nightmare and hell difficulty, because the base item selected is based on the adjusted monster level. Then they gave base items intrinsic affixes, and the best of those items (base items with haste built in, for example), could only be found in hell difficulty.

**Reply by toastmartin** [reply]
@StephenCWills: If I understand Jarulf right, the dropped item would be itemlevel 60. The type of the base item would be 1-60 and the qlevel of the pre/suffixes would be 32-64. Staves could choose from wyrm, hydra and champion. There are options for every item type except Jewlery. But you are right, it could be boring to see all the items from higher monsters just choosing from a hand full of properties. I understand that either the formulas for item generation or the complete table of qlevels would need an overhaul. For example the minimum qlevel could be set to 25.

@DakkJaniels: At the moment you can obtain the best items in game from visiting vendors with the right items in your inventory. Killing monsters for those items would definitely be an upgrade, even if they are not the hardest in the game. But you are right, that my proposed change is not that simple and requires more changes.

**Reply by StephenCWills** [reply]
> If I understand Jarulf right, the dropped item would be itemlevel 60. The type of the base item would be 1-60 and the qlevel of the pre/suffixes would be 32-64. Staves could choose from wyrm, hydra and champion. There are options for every item type except Jewlery. But you are right, it could be boring to see all the items from higher monsters just choosing from a hand full of properties. I understand that either the formulas for item generation or the complete table of qlevels would need an overhaul. For example the minimum qlevel could be set to 25.

Perhaps you understand Jarulf right, but you don't quite understand item generation fully enough to do such a deep dive into a hypothetical situation that Jarulf never had to account for.

First, there's actually a minimum qlvl of 25 that Jarulf would have likely ignored because ilvl 30 was a practical limit for his guide. So an ilvl 60 item from Lazarus would actually get affixes from 25-64. Unfortunately, staves that can have spells (such as those dropped from monsters in the dungeon) have some additional logic that overrides the minimum qlvl so you would be correct to say that the affix pool for the staff only includes affixes with qlvl between 32-64, but you would be incorrect to say that jewelry would have no affixes to choose from.

Anyway, back to the staff example. Wyrm's, Hydra's, and Champion's are all prefixes. If a magical staff rolled with only a suffix and no prefix, there would be no suffix that the game could assign to the staff. Item generation would not backtrack to replace the missing suffix with a valid prefix, but rather would simply fail to assign a suffix. FYI, I ran the test myself a long while back and managed to get a non-magical staff to drop from Lazarus this way so I'm not just speculating.
